api/app/Http/Controllers/ChildEntityController.php

```diff
@@ -9,22 +9,51 @@
 namespace App\Http\Controllers;
 
 use App\Extensions\Controller\RequestValidationTrait;
+use Illuminate\Database\Eloquent\Builder;
 use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Illuminate\Database\Eloquent\Relations\HasOneOrMany;
-use Illuminate\Database\Eloquent\Relations\Relation;
 use Illuminate\Http\Request;
-use Spira\Repository\Model\BaseModel;
-use Spira\Repository\Validation\ValidationException;
+use Spira\Model\Collection\Collection;
+use Spira\Model\Model\BaseModel;
+use Spira\Model\Validation\ValidationException;
+use Spira\Model\Validation\ValidationExceptionCollection;
+use Spira\Responder\Contract\TransformerInterface;
 use Spira\Responder\Response\ApiResponse;
 
 class ChildEntityController extends ApiController
 {
     use RequestValidationTrait;
 
-    protected $validateChildRequestRule = 'uuid';
-
+    protected $validateParentIdRule = null;
+    protected $validateChildIdRule = null;
     protected $relationName = null;
 
+    /**
+     * @var BaseModel
+     */
+    protected $cacheChildModel;
+
+    /**
+     * @var BaseModel
+     */
+    protected $parentModel;
+
+    public function __construct(BaseModel $parentModel, TransformerInterface $transformer)
+    {
+        $this->parentModel = $parentModel;
+
+        if (!$this->relationName) {
+            throw new \InvalidArgumentException('You should specify relationName in '.static::class);
```

Comment:
Phrasing of the exception should be `You *must* specify[...]` the word `should` implies it is optional.